### PR TITLE
Rework access logic

### PR DESCRIPTION
**Pull request by IvanIvanoff**
#### Summary
For a customer, we want to add exchange wallets to their plan. The solution is a generic one: In order to allow access to some queries only when a user has a certain plan (extension plan) you need to do the following:
- Add a new Product and define a Plan with it.
- Add this plan to the customer stripe account.
- The GraphQL query definition must be enriched with the following meta: 
```elixir
meta(access: :extension, product: Product.product_exchange_wallets())
```
where the product is the needed product. In order to pass through the checks for this query, the `AccessControl` middleware must be added, too.

[//]: # (#### Related PRs)

[//]: # (#### Additional deploy notes)

[//]: # (#### Screenshots)

### CODE REVIEW

File: lib/sanbase_web/graphql/middlewares/access_control.ex [hunk 1]
```diff
@@ -64,18 +72,38 @@ defmodule SanbaseWeb.Graphql.Middlewares.TimeframeRestriction do
     resolution
   end
 
+  @exchange_wallets_product Product.product_exchange_wallets()
+  @plan_extension_needed [:exchange_wallets]
```
Comment (tspenov) [inline lib/sanbase_web/graphql/middlewares/access_control.ex]: Shouldn't we fill this list from query meta tags: `meta(access: :extension)` :?

Comment (IvanIvanoff) [reply lib/sanbase_web/graphql/middlewares/access_control.ex]: It's a POC - currently I'm doing a major rework on this part

Comment (IvanIvanoff) [reply lib/sanbase_web/graphql/middlewares/access_control.ex]: The end result will be something like this: 
image

Comment (IvanIvanoff) [reply lib/sanbase_web/graphql/middlewares/access_control.ex]: @tspenov it's reworked - you can check it again.

File: lib/sanbase/web.ex [hunk 2]
```diff
@@ -19,7 +19,7 @@ defmodule Sanbase.Application.Web do
       Sanbase.TimescaleRepo,
 
       # Start the Clickhouse Repo
-      start_in({Sanbase.ClickhouseRepo, []}, [:dev, :prod]),
```
Comment (tspenov) [inline lib/sanbase/web.ex]: leftover

File: lib/sanbase/billing/graphql_schema.ex [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+      {{_, _, [module, func]}, _, _} = product_ast
+      product_id = apply(module, func, [])
+      {f, product_id}
+    end)
+    |> Map.new()
+  end
+
+  @doc ~s"""
+  Return all query names that have all `fields` with the values specified in
+  the corresponding position of the `values` lis
```
Comment (tspenov) [inline lib/sanbase/billing/graphql_schema.ex]:
```suggestion
  the corresponding position of the `values` list
```

**Pull request merged**